lib/unleash/constraint.rb

```diff
@@ -55,9 +55,9 @@ def matches_context?(context)
 
     def self.on_valid_date(val1, val2)
       val1 = DateTime.parse(val1)
-      val2 = DateTime.parse(val2)
+      val2 = val2.is_a?(DateTime) ? val2 : DateTime.parse(val2)
       yield(val1, val2)
-    rescue ArgumentError
+    rescue ArgumentError, TypeError, Date::Error
```

Comment:
Note that `Date:Error` inherits from ArgumentError, so it should be dropped as it is rescued already:

```suggestion
    rescue ArgumentError, TypeError
```